pkg/kubectl/bind/plugin/bind.go

```diff
@@ -207,101 +248,105 @@ func (b *BindOptions) Run(ctx context.Context, urlCh chan<- string) error {
 	}
 	var secretName string
 	for _, s := range secrets.Items {
-		if s.Annotations[resources.ClusterIDAnnotationKey] == response.ID {
+		if s.Annotations[resources.ClusterIDAnnotationKey] == authResponse.ID {
 			secretName = s.Name
 			break
 		}
 	}
 
 	// check for existing CRD
-	crd, err := apiextensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, response.Resource+"."+response.Group, metav1.GetOptions{})
-	if err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
-	if secretName == "" {
-		if err == nil {
-			return fmt.Errorf("CRD %s.%s already exists and is not from this service provider", response.Resource, response.Group)
-		}
-
-		fmt.Fprintf(b.IOStreams.Out, "Creating secret for identity %s\n", response.ID) // nolint: errcheck
-		secretName, err = resources.EnsureServiceBindingAuthData(ctx, string(response.Kubeconfig), response.ID, "kube-bind", "", kubeClient)
-		if err != nil {
-			return err
-		}
-	} else if err == nil {
-		fmt.Fprintf(b.IOStreams.Out, "Found existing CRD %s. Checking owner.\n", crd.Name) // noline: errcheck
-
-		// check if the CRD is owner-refed by the APIServiceBinding
-		for _, ref := range crd.OwnerReferences {
-			parts := strings.SplitN(ref.APIVersion, "/", 2)
-			if parts[0] != kubebindv1alpha1.SchemeGroupVersion.Group || ref.Kind != "APIServiceBinding" {
-				continue
-			}
-
-			existing, err := bindClient.KubeBindV1alpha1().APIServiceBindings().Get(ctx, ref.Name, metav1.GetOptions{})
+	for _, request := range apiRequests {
+		for _, resource := range request.Spec.Resources {
```

Comment:
follow-up: split into separate func, this gets big

Reply:
This will be split into the two commands.

But even after that, this logic needs unit tests.